ILSpy.ReadyToRun/ReadyToRunLanguage.cs

```diff
@@ -103,12 +105,92 @@ public override void WriteCommentLine(ITextOutput output, string comment)
+								case VarLocType.VLT_REG_FP:
+								case VarLocType.VLT_REG_BYREF:
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+
+								case VarLocType.VLT_STK:
+								case VarLocType.VLT_STK_BYREF:
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data2}");
+									break;
+								case VarLocType.VLT_REG_REG:
+									output.WriteLine($"    Register 1: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Register 2: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									break;
+								case VarLocType.VLT_REG_STK:
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data3}");
+									break;
+								case VarLocType.VLT_STK_REG:
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data1}");
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data3)}");
+									break;
+								case VarLocType.VLT_STK2:
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data2}");
+									break;
+								case VarLocType.VLT_FPSTK:
+									output.WriteLine($"    Offset: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+								case VarLocType.VLT_FIXED_VA:
+									output.WriteLine($"    Offset: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+
+								default:
+									throw new BadImageFormatException("Unexpected var loc type");
+							}
+
+							output.WriteLine("");
+						}
+					}
 
+				}
+			}
+			return debugInfoDict;
+		}
 
-		private Dictionary<ulong, UnwindCode> WriteUnwindInfo(RuntimeFunction runtimeFunction, ITextOutput output)
+
+		private Dictionary<ulong, HashSet<UnwindCode>> WriteUnwindInfo(RuntimeFunction runtimeFunction, ITextOutput output)
 
 		{
-			Dictionary<ulong, UnwindCode> unwindCodes = new Dictionary<ulong, UnwindCode>();
+			Dictionary<ulong, HashSet<UnwindCode>> unwindCodes = new Dictionary<ulong, HashSet<UnwindCode>>();
```

Comment:
for System.io.FileLoadException.GetMessageForHR in System.private.corelib I am finding 2 opcodes with the same offset of 19

Reply:
The case is understood, the root cause is a [bug ](https://github.com/dotnet/runtime/issues/39940)in `ILCompiler.Reflection.ReadyToRun`. It would be great that we fix the root cause instead of having this hack.

Reply:
This [PR](https://github.com/dotnet/runtime/pull/40030) should fix it.